src/lib.rs

```diff
@@ -255,8 +298,11 @@ impl Executor {
             let (lock, res) = self.cvar.wait_timeout(inner, timeout).unwrap();
             inner = lock;
 
-            // If there are no tasks after a while, stop this thread.
-            if res.timed_out() && inner.queue.is_empty() {
+            // If there are too many threads active in the pool, stop this thread.
+            if (Some(inner.thread_count) > inner.thread_limit.map(NonZeroUsize::get))
+                // If there are no tasks after a while, stop this thread.
+                && (res.timed_out() && inner.queue.is_empty())
+            {
```

Comment:
My immediate concern here is that there's a possible race condition, where every thread observes that there are too many threads and shrinks the pool to a size smaller than zero. Any chance we can have an explicit check to prevent this, or an explicit `grow_pool`?

Reply:
Isn't this under a lock? Wouldn't this decrease one by one?